### ISSUE TITLE
Make Gitify windows compatible

### ISSUE THREAD

**Issue by vguna** [modmore/Gitify#106]
Hi.

Although it is mentioned in the Wiki, Gitify isn't completely runnable under Windows.

E.g. Gitify modx:install needs at least:

- curl
- mv
- rm
- ls
- grep
- head

It should be easy to replace these linux native calls with an appropriate php function, I guess :)?

**Comment by Mark-H** [comment]
My experience developing on windows is fairly limited and definitely outdated as I've been using Macs for the past 4 years or so, but those are pretty standard commands for a terminal/command prompt? Which one(s) is(are) not installed for you, causing what errors?

**Comment by vguna** [comment]
All of them :). Windows does not provide any of those commands. A shame, I know :).
But I think each of these commands could be replaced by a simple php call, couldn't it?
This way it would be portable across all php supported platforms.

**Comment by Mark-H** [comment]
Are you using cmd.exe or a slightly more decent command prompt like Git Bash (msysgit)?

Some of those things can definitely be handled with some more code with php, but I'm kinda failing to see how any terminal that doesn't have even something basic like `ls` has a place in the development world :P

**Comment by vguna** [comment]
cmd.exe - "dir" would be the synonym for "ls" on linux - but parameters are different of course :).

**Comment by Thunderofnl** [comment]
I agree with @Mark-H CMD doesn't belong in Gitify. You can get gitify to work by installing git bash (Which should be included in your git installation) and using that instead of cmd. It's a small terminal emulator that has all the necessary functions.

**Comment by vguna** [comment]
Don't get me wrong - it isn't about how to get Gitify working under Windows :). That works as you describe. It's more about portability and whether one should use native OS calls although that could be done with PHP itself. It would simply reduce/remove the dependencies on the OS system.

Also in my case, I don't use git at all. I use subversion and Gitify works with that too. In that case the git binaries aren't installed. Although GITify sounds like git is mandatory, it isn't - until now ;). And the gitify documentation is using the term SCM instead of git, and that is the right approach :).

Maybe If find time to contribute a PR that removes the OS dependencies.

**Comment by Mark-H** [comment]
Gotcha. For certain things calling the shell commands seemed like a better approach at the time, but definitely open to making things more portable. 

One thing I'd probably try to avoid replacing is the curl command - it has a nice progress indicator that I don't think will be easily replaceable with pure PHP code.

**Comment by siteleas** [comment]
Here are the Windows dependencies in order to get the script to execute properly. They must be installed and their bin folders have to be added to the Windows PATH environment variable in the Advanced System Settings screen:

* compatible unzip capabilities - http://iweb.dl.sourceforge.net/project/gnuwin32/unzip/5.51-1/unzip-5.51-1-bin.zip
* compatible linux bsh commands - http://superb-dca2.dl.sourceforge.net/project/win-bash/shell-complete/latest/shell.w32-ix86.zip
* compatible curl support - http://www.confusedbycode.com/curl/#downloads
* curl needs the msvc redistributable in order to function (install this before curl) - https://download.microsoft.com/download/9/3/F/93FCF1E7-E6A4-478B-96E7-D4B285925B00/vc_redist.x64.exe
* windows composer support - https://getcomposer.org/Composer-Setup.exe

Once those are completed, the rest of the instructions do work.

I did run into a snag with the Gitify init feature though. MODx seems to install and setup correctly but then some of the MySQL tables are missing (such as modx_session, modx_content_type, modx_access_policies, modx_access_policies)

Here is the error.log file:
    ```
[2016-01-14 07:46:42] (ERROR @ /index.php) Error 42S02 executing statement: 
Array
(
    [0] => 42S02
    [1] => 1146
    [2] => Table 'businessclubs_ca.modx_session' doesn't exist
)

[2016-01-14 07:46:42] (ERROR @ /index.php) Error 42S02 executing statement: 
Array
(
    [0] => 42S02
    [1] => 1146
    [2] => Table 'businessclubs_ca.modx_access_policies' doesn't exist
)

[2016-01-14 07:46:42] (ERROR @ /index.php) Error 42S02 executing statement: 
Array
(
    [0] => 42S02
    [1] => 1146
    [2] => Table 'businessclubs_ca.modx_content_type' doesn't exist
)

[2016-01-14 07:46:42] (FATAL) The requested resource has no valid content type specified.
[2016-01-14 07:46:42] (ERROR @ /index.php) Error 42S02 executing statement: 
Array
(
    [0] => 42S02
    [1] => 1146
    [2] => Table 'businessclubs_ca.modx_session' doesn't exist
)

[2016-01-14 07:46:42] (ERROR @ /index.php) Error 42S02 executing statement:
INSERT INTO `modx_session` (`id`, `access`, `data`) VALUES ('1u9q6ej65bat0bkff8ni3nnl42', 1452757602, '')
Array
(
    [0] => 42S02
    [1] => 1146
    [2] => Table 'businessclubs_ca.modx_session' doesn't exist
)
    ```
